### DISCUSSION TITLE
How to replace an image when clicking on a table row?

### DISCUSSION THREAD

**Discussion by dontlagother** [zauberzeug/nicegui#2346] [Q&A] [answered]
### Question

How does nicegui dynamically change the content of another component when setting the table click event, such as using the ui.image component to display the corresponding image for each row when I click on a row of the table

**Comment by falkoschindler** [comment]
Hi @dontlagother, please be more specific: What have you already tried? Where exactly do you struggle? Try to break it down into parts: showing a table, handling row click events, replacing an image. For each of these parts there should be demos in the documentation. If this doesn't help, it's best to post a minimal reproducible code example of the specific problem you're facing.

**Reply by dontlagother** [reply]
I've already made a few attempts to toggle an image that appears below the existing layout instead of where I specified it using the function click event.
I tried using global variables to solve this problem but it didn't work.
Here's the main code:
<img width="856" alt="屏幕截图 2024-01-11 172055" src="https://github.com/zauberzeug/nicegui/assets/90087852/88558c19-d89c-4a95-877d-01ee7dd06fc9">
<img width="431" alt="屏幕截图 2024-01-11 172119" src="https://github.com/zauberzeug/nicegui/assets/90087852/633e3443-123a-4337-9194-6557833e3ccc">

**Reply by falkoschindler** [reply, marked as answer]
Instead of creating a new image via `ui.image()`, you might want to call `Image.source = ...` or `Image.set_source(...)`.

**Reply by dontlagother** [reply]
Thank you for your answer. I'll try this solution.

**Reply by dontlagother** [reply]
In addition, I would like to ask if there is a similar method in ui.table, that is, clicking on a row of a table can dynamically change the content of another table row.

**Reply by falkoschindler** [reply]
Besides `.add_rows()` there is no dedicated method for manipulating the rows.
But you can simply update the table's `rows` list and call `.update()` afterwards.

**Reply by dontlagother** [reply]
Thank you very much for your answer.
